assets/js/frontend/checkout.js

```diff
@@ -303,6 +317,21 @@ jQuery( function( $ ) {
 						$( '#terms' ).prop( 'checked', true );
 					}
 
+					// Fill in the payment details if possible
+					if ( ! $.isEmptyObject( paymentDetails ) ) {
+						$( '.payment_box input' ).each( function() {
+							var ID = $( this ).attr( 'id' );
+
+							if ( ID ) {
+								if ( 'checkbox' === $( this ).attr( 'type' ) && paymentDetails[ ID ] ) {
+									$( this ).prop( 'checked', true );
+								} else {
+									$( this ).val( paymentDetails[ ID ] );
```

Comment:
Trigger change() after these so any logic that hooks into change is fired. E.g. token generation.

Can you test with Stripe?